src/seedu/addressbook/AddressBook.java

```diff
@@ -866,13 +866,15 @@ private static String getEmailFromPerson(String[] person) {
      * @param name of person
      * @param phone without data prefix
      * @param email without data prefix
+     * @param dob 
      * @return constructed person
      */
-    private static String[] makePersonFromData(String name, String phone, String email) {
+    private static String[] makePersonFromData(String name, String phone, String email, String dob) {
         final String[] person = new String[PERSON_DATA_COUNT];
         person[PERSON_DATA_INDEX_NAME] = name;
         person[PERSON_DATA_INDEX_PHONE] = phone;
         person[PERSON_DATA_INDEX_EMAIL] = email;
+        person[3] = dob;
```

Comment:
Magic number here although you refactored the rest of the numbers for name, phone and email.